Middleware-with-Tor/tor/tor/middlewares.py

```diff
@@ -0,0 +1,61 @@
+import sys
+import os
+from dotenv import load_dotenv
+from stem import Signal
+from stem.control import Controller
+from scrapy.downloadermiddlewares.httpproxy import HttpProxyMiddleware
+from stem.util.log import get_logger
+
+sys.path.append(os.path.abspath(".."))
```

Comment:
Why did you need this?

Reply:
Because python wasn't setting the `settings.py` module, so I had to add its path so I can import `settings.py`.

Reply:
Fixed.